### PR TITLE
modulegen: create internal/mkdocs

### PR DESCRIPTION
**Pull request by mmorel-35**
Refactor mkdocs code in modulegen

### CODE REVIEW

File: modulegen/main.go [hunk 1]
```diff
@@ -375,3 +345,12 @@ func runGoCommand(cmdDir string, args ...string) error {
 	cmd.Dir = cmdDir
 	return cmd.Run()
 }
+
+func getRootDir() (string, error) {
```
Comment (mdelapenya) [inline modulegen/main.go:349]: I think we can have a `NewContext(dir string) (Context, error)` function in the `context.go` file, instead of this one. In the end, we are always using it to get the dir, assert there is no error and create a context.

**Review by mdelapenya — DISMISSED** [review]
LGTM!, just a comment, not a blocker, regarding one function. Other than that, great job extracting code! 💪

File: modulegen/main.go (hunk 1 above)
Comment (mmorel-35) [reply modulegen/main.go:349]: Do you want this in this PR ?
I can add this in the new one as it will be a small one.

Comment (mdelapenya) [reply modulegen/main.go:349]: I think it's fine to have this change in this PR, as it's very narrowed to the PR itself. So I'd prefer having it here if possible 🙏

Comment (mmorel-35) [reply modulegen/main.go:347]: I have created NewContext() but for the moment I cannot remove getRootDir as getModulesOrExamples still relies on it.
It shall be easier once getModulesOrExamples is replaced by ctx.GetModules()

Comment (mmorel-35) [reply modulegen/main.go:347]: @mdelapenya ,
Can you please trigger a retry on the workflow ?

**Comment by mmorel-35** [comment]
By the way, I noticed that there is no `docs/modules/compose.md` is it on purpose ?

**Comment by mdelapenya** [comment]
> By the way, I noticed that there is no `docs/modules/compose.md` is it on purpose ?

Indeed, it's on purpose as we do have a separate page in https://golang.testcontainers.org/features/docker_compose/.

But I think it would be a good idea to add it to the modules and reduce the number of corner cases across code generation. I'll do it in a separate PR. Thanks for pointing this out!

File: modulegen/context.go [hunk 2]
```diff
@@ -45,6 +45,25 @@ func (ctx *Context) getModulesByBaseDir(baseDir string) ([]string, error) {
 	return dirs, nil
 }
 
+func (ctx *Context) getMardownsByBaseDir(baseDir string) ([]string, error) {
```
Comment (mdelapenya) [inline modulegen/context.go:48]:
```suggestion
func (ctx *Context) getMarkdownsFromDir(baseDir string) ([]string, error) {
```

File: modulegen/context.go [hunk 3]
```diff
@@ -53,6 +72,18 @@ func (ctx *Context) GetModules() ([]string, error) {
 	return ctx.getModulesByBaseDir("modules")
 }
 
+func (ctx *Context) GetExamplesDocs() ([]string, error) {
+	return ctx.getMardownsByBaseDir("examples")
```
Comment (mdelapenya) [inline modulegen/context.go:76]:
```suggestion
	return ctx.getMarkdownsFromDir("examples")
```

File: modulegen/context.go [hunk 4]
```diff
@@ -53,6 +72,18 @@ func (ctx *Context) GetModules() ([]string, error) {
...
+}
+
+func (ctx *Context) GetModulesDocs() ([]string, error) {
+	return ctx.getMardownsByBaseDir("modules")
```
Comment (mdelapenya) [inline modulegen/context.go:80]:
```suggestion
	return ctx.getMarkdownsFromDir("modules")
```

**Review by mdelapenya — APPROVED** [review]
LGTM! Only a final change in a method name: there was a typo and making the name a bit more consistent.

Other than that, once this is addressed, we are ready to merge

**Comment by mmorel-35** [comment]
@mdelapenya ,
I just changed the method name.
The tests are running.

**Comment by mmorel-35** [comment]
Hi @mdelapenya ,

The tests were passing before the rebase, something happened with the two last commits on the main branch ?

**Comment by mdelapenya** [comment]
> The tests were passing before the rebase, something happened with the two last commits on the main branch ?

Not necessarily with the two commit :) I always take a look at the workflow:

Screenshot 2023-08-24 at 08 50 28

Because we run against the last two active versions of Go, I check both on Ubuntu: if one passes and the other not, then it seems certain flakiness to me. Same for the modules.

File: modulegen/main.go [hunk 5]
```diff
@@ -170,14 +171,13 @@ func main() {
 	fmt.Println("Thanks!")
 }
 
-func generate(example Example, rootDir string) error {
+func generate(example Example, ctx *Context) error {
```
Comment (mdelapenya) [inline modulegen/main.go:174]: I did not notice this: for consistency, let's pass the context as first arg.

But let's do this in a separate PR in order to merge this one as is :)

**Review by mdelapenya — APPROVED** [review]
LGTM, thanks for your hard work adding maintainability to the code generation tool 👏
